pkg/network/usm/monitor_test.go

```diff
@@ -991,6 +991,54 @@ func generateTestHeaderFields(pathNeverIndexed, withoutIndexing bool) []hpack.He
 	}
 }
 
+// removeHeaderFieldByKey removes the header field with the given key from the given header fields.
+func removeHeaderFieldByKey(headerFields []hpack.HeaderField, keyToRemove string) []hpack.HeaderField {
+	for i := 0; i < len(headerFields); i++ {
+		if headerFields[i].Name == keyToRemove {
+			// Remove the element by modifying the slice in-place
+			headerFields = append(headerFields[:i], headerFields[i+1:]...)
+			break
+		}
+	}
+
+	return headerFields
+}
+
+// createHeadersWithIndexedPathKey creates a message with the given number of header frames but the
+// path frame header key 4 and not 5 as usual.
+func createHeadersWithIndexedPathKey(t *testing.T, headerFields []hpack.HeaderField, headersCount int) []byte {
+	var buf bytes.Buffer
+	framer := http2.NewFramer(&buf, nil)
+	// pathHeaderField is the hex representation of the path /aaa with index 4.
+	pathHeaderField := []byte{0x44, 0x83, 0x60, 0x63, 0x1f}
+
+	// we remove the header field with key ":path" so that we will not have two header fields.
+	headerFields = removeHeaderFieldByKey(headerFields, ":path")
+
+	for i := 0; i < headersCount; i++ {
+		streamID := 2*i + 1
+		headersFrame, err := usmhttp2.NewHeadersFrameMessage(headerFields)
+		require.NoError(t, err, "could not create headers frame")
+
+		// we are adding the path header field with index 4, we need to do it on the byte slice and not on the headerFields
+		// due to the fact that when we create a header field it would be with index 5.
+		headersFrame = append(pathHeaderField, headersFrame...)
```

Comment:
You're adding a path header on every header

Reply:
similar to https://github.com/DataDog/datadog-agent/pull/21909/files#r1453371149

Reply:
The change should solve the confusion over here